--- a/src/Database/TenantScope.php
+++ b/src/Database/TenantScope.php
@@ -17,7 +17,7 @@
             return;
         }
 
-        $builder->where($model->qualifyColumn(BelongsToTenant::$tenantIdColumn), tenant()->getTenantKey());
+        $builder->where($model->qualifyColumn($model::$tenantIdColumn), tenant()->getTenantKey());
     }
 
     public function extend(Builder $builder)
--- a/src/Database/Concerns/BelongsToTenant.php
+++ b/src/Database/Concerns/BelongsToTenant.php
@@ -16,7 +16,7 @@
 
     public function tenant()
     {
-        return $this->belongsTo(config('tenancy.tenant_model'), BelongsToTenant::$tenantIdColumn);
+        return $this->belongsTo(config('tenancy.tenant_model'), static::$tenantIdColumn);
     }
 
     public static function bootBelongsToTenant()
@@ -24,9 +24,9 @@
         static::addGlobalScope(new TenantScope);
 
         static::creating(function ($model) {
-            if (! $model->getAttribute(BelongsToTenant::$tenantIdColumn) && ! $model->relationLoaded('tenant')) {
+            if (! $model->getAttribute(static::$tenantIdColumn) && ! $model->relationLoaded('tenant')) {
                 if (tenancy()->initialized) {
-                    $model->setAttribute(BelongsToTenant::$tenantIdColumn, tenant()->getTenantKey());
+                    $model->setAttribute(static::$tenantIdColumn, tenant()->getTenantKey());
                     $model->setRelation('tenant', tenant());
                 }
             }
